tests/test_app_init_rebinding_spec.py

```
@@ -0,0 +1,152 @@
+        assert app.__spec__.name == "app"
+
+    def test_rebinding_spec_name_attribute_access(self):
+        """Test that accessing spec.name restores sys.modules binding."""
+        import app
+
+        # Save original binding
+        original_module = sys.modules.get("app")
+        assert original_module is not None
+
+        # Access spec.name (this should trigger rebinding logic)
+        spec = getattr(app, "__spec__")
```

Ruff (0.13.1)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)